PWG/EMCAL/EMCALtasks/AliEmcalCorrectionClusterNonLinearityMCAfterburner.h

```diff
@@ -0,0 +1,62 @@
+#ifndef ALIEMCALCORRECTIONCLUSTERNONLINEARITYMCAFTERBURNER_H
+#define ALIEMCALCORRECTIONCLUSTERNONLINEARITYMCAFTERBURNER_H
+
+#include "AliEmcalCorrectionComponent.h"
+
+/**
+ * @class AliEmcalCorrectionClusterNonLinearityMCAfterburner
+ * @ingroup EMCALCOREFW
+ * @brief Cluster energy non-linearity correction MC afterburner in the EMCal correction framework.
+ *
+ * Non-linearity correction to the MC cluster energy is necessary because the current non linearity is done for
+ * 50MeV cell threshold and all analyses use a 100MeV cell threshold for the clusterization.
+ * Thus, the MC need to be corrected in a second step
+ 
+ The energy of the cluster **after** the MC non-linearity correction can be retrieved using the method `cluster->GetNonLinCorrEnergy()`.
+ *
+ * Based on code in AliCaloPhotonCuts.cxx.
+ *
+ * @author 15 active people originally in AliCaloPhotonCuts.cxx
+ * @author Eliane Epple, Yale University, include the PCM aproach into the EMCal correction FW using correction components
+ * @date Nov 13, 2018
+ */
+
+
+class AliEmcalCorrectionClusterNonLinearityMCAfterburner : public AliEmcalCorrectionComponent {
+ public:
+
+  AliEmcalCorrectionClusterNonLinearityMCAfterburner();
+  virtual ~AliEmcalCorrectionClusterNonLinearityMCAfterburner();
+
+  // Sets up and runs the task
+  Bool_t Initialize();
+  void UserCreateOutputObjects();
+  Bool_t Run();
+  TString SummarizeMCProductions(TString namePeriod);
+  void InitNonLinearityParam(TString namePeriod, Int_t version);
+
+protected:
+  TH1F                  *fEnergyDistBefore;          			//!<!energy distribution before
+  TH2F                  *fEnergyTimeHistBefore;      			//!<!energy/time distribution before
+  TH1F                  *fEnergyDistAfter;          	  		  //!<!energy distribution after
+  TH2F                  *fEnergyTimeHistAfter;       			//!<!energy/time distribution after
+  Float_t                fNLAfterburnerPara[9];      			///< Parameters for the non linearity function
+  Int_t                  fNonLinearityAfterburnerFunction; ///< Type of function used for the non linearity afterburner correction
+
+  Int_t                  fAfterburnerVersion;        			///< The version of the non-linearity afterburner correction (0-3)
+  TString                fMCPeriod;                  			///< The MC production name
+  Bool_t                 fSetForceClusterE;          			///< Only for backwards compatibility, force cluster->E() to be set to the cluster non-linearity corrected energy. Off by default. For the standard methods, see: http://alidoc.cern.ch/AliPhysics/master/READMEcontainers.html#emcalContainerClusterEnergyCorrections
```

Comment:
I think this is to be backwards compatible with the very old way of handling things. That you do not have new entries for the modified cluster energy but you only modify the main energy field. But its also true that this is a new component. I think the only hypothetical case where this could get relevant is when someone has still a very old code that they are using and then want to incorporate this new correction.

Reply:
Okay, thanks. I would vote that we don't really need it, but I guess it also doesn't really hurt anything. So it's up to you and others if you want to keep/remove it

Reply:
I would also vote to remove it, but up to you